packages/block-editor/src/store/selectors.js

```diff
@@ -958,11 +958,15 @@ export function __unstableIsSelectionCollapsed( state ) {
 	);
 }
 
+export function __unstableIsBlockMergeable( state, clientId ) {
+	const blockName = getBlockName( state, clientId );
+	const blockType = getBlockType( blockName );
+	return !! blockType.merge;
```

Comment:
The only way for `blockType` to be empty is if the selector is called with a bogus `clientId`, but it doesn't hurt to be defensive:

```suggestion
	return !! blockType?.merge;
```